src/test/java/jewellery/inventory/integration/AuthIntegrationTest.java

```diff
@@ -0,0 +1,127 @@
+package jewellery.inventory.integration;
+
+import static jewellery.inventory.helper.UserTestHelper.USER_EMAIL;
+import static jewellery.inventory.helper.UserTestHelper.USER_NAME;
+import static jewellery.inventory.helper.UserTestHelper.USER_PASSWORD;
+import static jewellery.inventory.helper.UserTestHelper.createTestUser;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.when;
+
+import jewellery.inventory.dto.request.AuthenticationRequestDto;
+import jewellery.inventory.dto.response.AuthenticationResponseDto;
+import jewellery.inventory.model.User;
+import jewellery.inventory.repository.UserRepository;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.crypto.password.PasswordEncoder;
+
+class AuthIntegrationTest extends AuthenticatedIntegrationTestBase {
+
+  @Autowired private UserRepository userRepository;
+  @MockBean private PasswordEncoder passwordEncoder;
```

Comment:
I think yes. Because we create a testUser and to save it properly in the mock database the password should be encoded using the encoder... If I am missing something let me know.

Reply:
but with the other integration tests the database isn't mocked. Are we mocking it here?

Reply:
I think for this we need valid (existing) user with valid credentials provided (existing email and existing encoded password) so it generates token. Will check if the process could be simplified